python_modules/libraries/dagster-airbyte/dagster_airbyte/managed/reconciliation.py

```diff
@@ -0,0 +1,510 @@
+                )
+            existing_source = None
+
+        if configured_source:
+            defn_id = check.not_none(
+                res.get_source_definition_by_name(configured_source.source_type, workspace_id)
+            )
+            base_source_defn_dict = {
+                "name": configured_source.name,
+                "connectionConfiguration": configured_source.source_configuration,
+            }
+            source_id = "TBD"
+            if existing_source:
+                source_id = existing_source.source_id
+                if not dry_run:
+                    res.make_request(
+                        endpoint="/sources/update",
+                        data={"sourceId": source_id, **base_source_defn_dict},
+                    )
+            else:
+                if not dry_run:
+                    create_result = cast(
+                        Dict[str, str],
+                        check.not_none(
+                            res.make_request(
+                                endpoint="/sources/create",
+                                data={
+                                    "sourceDefinitionId": defn_id,
+                                    "workspaceId": workspace_id,
+                                    **base_source_defn_dict,
+                                },
+                            )
+                        ),
+                    )
+                    source_id = create_result["sourceId"]
+
+            initialized_sources[source_name] = InitializedAirbyteSource(
+                source=configured_source,
+                source_id=source_id,
+                source_definition_id=defn_id,
+            )
+
+    return initialized_sources, diff
+
+
+def reconcile_destinations(
+    res: AirbyteResource,
+    config_destinations: Mapping[str, AirbyteDestination],
+    existing_destinations: Mapping[str, InitializedAirbyteDestination],
+    workspace_id: str,
+    dry_run: bool,
+) -> Tuple[Mapping[str, InitializedAirbyteDestination], ManagedElementCheckResult]:
+    """
+    Generates a diff of the configured and existing destinations and reconciles them to match the
+    configured state if dry_run is False.
+    """
+
+    diff = ManagedElementDiff()
+
+    initialized_destinations = {}
+    for destination_name in set(config_destinations.keys()).union(existing_destinations.keys()):
+        configured_destination = config_destinations.get(destination_name)
+        existing_destination = existing_destinations.get(destination_name)
+
+        diff = diff.join(
+            diff_destinations(
+                configured_destination,
+                existing_destination.destination if existing_destination else None,
+            )
+        )
+
+        if not configured_destination and existing_destination:
+            initialized_destinations[destination_name] = existing_destination
+            if not dry_run:
+                res.make_request(
```

Comment:
this part is a bit scary -- are we confident that all users of this interface will want Dagster to manage all of their Airbyte connections? Worried about a poor user who hooks this stuff up to their prod airbyte instance to try it out and it deletes a bunch of stuff automatically.

I'd prefer to just leave the irrelevant connections alone or put this behavior behind a flag or something if possible. Willing to hear other opinions though

Reply:
That seems reasonable to me. I'll add a flag to the reconciler object (defaults to off) that will tell it to delete sources/destinations/connections.